DeusEx/Classes/DXRando.uc

```diff
@@ -32,235 +22,86 @@ var int doorspickable, doorsdestructible, deviceshackable, passwordsrandomized,
-            SaveFlags();
-
-        RandoEnter();
-	}
-
+    if( Player == None ) {
+        PostPostBeginPlay();
+        return;
+    }
     CheckNotes();
 }
 
-// ----------------------------------------------------------------------
-// GetPatrolPoint()
-// ----------------------------------------------------------------------
-
-function PatrolPoint GetPatrolPoint(Name patrolTag, optional bool bRandom)
-{
-	local PatrolPoint aPoint;
-
-	aPoint = None;
-
-	foreach AllActors(class'PatrolPoint', aPoint, patrolTag)
-	{
-		if (bRandom && (FRand() < 0.5))
-			break;
-		else
-			break;
-	}
-
-	return aPoint;
-}
-
-// ----------------------------------------------------------------------
-// GetSpawnPoint()
-// ----------------------------------------------------------------------
-
-function SpawnPoint GetSpawnPoint(Name spawnTag, optional bool bRandom)
-{
-	local SpawnPoint aPoint;
-
-	aPoint = None;
-
-	foreach AllActors(class'SpawnPoint', aPoint, spawnTag)
-	{
-		if (bRandom && (FRand() < 0.5))
-			break;
-		else
-			break;
-	}
-
-	return aPoint;
-}
-
 function LoadFlags()
 {
-    seed = flags.GetInt('Rando_seed');
-
-    flagsversion = flags.GetInt('Rando_version');
-    brightness = flags.GetInt('Rando_brightness');
-    minskill = flags.GetInt('Rando_minskill');
-    maxskill = flags.GetInt('Rando_maxskill');
-    ammo = flags.GetInt('Rando_ammo');
-    multitools = flags.GetInt('Rando_multitools');
-    lockpicks = flags.GetInt('Rando_lockpicks');
-    biocells = flags.GetInt('Rando_biocells');
-    medkits = flags.GetInt('Rando_medkits');
-    speedlevel = flags.GetInt('Rando_speedlevel');
-    keysrando = flags.GetInt('Rando_keys');
-    doorspickable = flags.GetInt('Rando_doorspickable');
-    doorsdestructible = flags.GetInt('Rando_doorsdestructible');
-    deviceshackable = flags.GetInt('Rando_deviceshackable');
-    passwordsrandomized = flags.GetInt('Rando_passwordsrandomized');
-    gibsdropkeys = flags.GetInt('Rando_gibsdropkeys');
+    local FlagBase f;
+    f = DeusExPlayer(GetPlayerPawn()).FlagBase;
+    log("DXRando LoadFlags()");
+    seed = f.GetInt('Rando_seed');
+
+    flagsversion = f.GetInt('Rando_version');
+    brightness = f.GetInt('Rando_brightness');
+    minskill = f.GetInt('Rando_minskill');
+    maxskill = f.GetInt('Rando_maxskill');
+    ammo = f.GetInt('Rando_ammo');
+    multitools = f.GetInt('Rando_multitools');
+    lockpicks = f.GetInt('Rando_lockpicks');
+    biocells = f.GetInt('Rando_biocells');
+    medkits = f.GetInt('Rando_medkits');
+    speedlevel = f.GetInt('Rando_speedlevel');
+    keysrando = f.GetInt('Rando_keys');
+    doorspickable = f.GetInt('Rando_doorspickable');
+    doorsdestructible = f.GetInt('Rando_doorsdestructible');
+    deviceshackable = f.GetInt('Rando_deviceshackable');
+    passwordsrandomized = f.GetInt('Rando_passwordsrandomized');
+    gibsdropkeys = f.GetInt('Rando_gibsdropkeys');
 
     if(flagsversion < 1) {
         log("DXRando upgrading flags from v"$flagsversion);
```

Comment:
move this log statement down into if(flagsversion < 2), it should follow the latest version